tests/basic/basic_test.go

```diff
@@ -316,6 +316,53 @@ var _ = Describe("{AppScaleUpAndDown}", func() {
 	})
 })
 
+var _ = Describe("{CordonDeployDestroy}", func() {
+	It("has to cordon all nodes but one, deploy and destroy app", func() {
+
+		Step("Cordon all nodes but one", func() {
+			nodes := node.GetWorkerNodes()
+			for _, node := range nodes[1:] {
+				err := Inst().S.DisableSchedulingOnNode(node)
+				Expect(err).NotTo(HaveOccurred())
+			}
+		})
+		var contexts []*scheduler.Context
+		Step("Deploy applications", func() {
+			for i := 0; i < Inst().ScaleFactor; i++ {
+				contexts = append(contexts, ScheduleAndValidate(fmt.Sprintf("cordondeploydestroy-%d", i))...)
+			}
+		})
+		Step("Destroy apps", func() {
+			opts := make(map[string]bool)
+			opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+			for _, ctx := range contexts {
+				err := Inst().S.Destroy(ctx, opts)
+				Expect(err).NotTo(HaveOccurred())
+			}
+		})
+		Step("Validate destroy", func() {
+			opts := make(map[string]bool)
+			opts[scheduler.OptionsWaitForResourceLeakCleanup] = true
+			for _, ctx := range contexts {
+				err := Inst().S.WaitForDestroy(ctx)
```

Comment:
what is the current timeout for waitfordestory? And is that per context?

Reply:
As discussed offline, let's fix the Destroy and WaitForDestroy timeouts.

Reply:
I was expecting WaitForDestroy to now take in a timeout.

Reply:
this is inside WaitForDestroy https://github.com/portworx/torpedo/pull/252/commits/dd929e3c9517bf5dbc6809a5217c3141b6a3277a

Reply:
The `defaultTimeout` inside WaitForDestroy won't work. The reason we added custom timeouts to the API was so that callers can use different timeouts based on their use case.

The timeout for WaitForDestroy needs to be proportional to number of apps inside ctx. We had seen that this test was timing out as we had pods at a larger scale.